src/runtime_src/core/tools/common/SmiWatchMode.h

```diff
@@ -95,4 +102,30 @@ class smi_watch_mode {
                  std::ostream& output,
                  const ReportGenerator& report_generator,
                  const std::string& report_title = "Report");
+
+  /**
+   * @brief Allocate and initialize a debug buffer for firmware log/trace data
+   * 
+   * @param log_buffer Reference to firmware_debug_buffer structure to initialize
+   * @param abs_offset Initial absolute offset for the buffer
+   * @param b_wait Whether the driver should wait for new events
+   * @return std::unique_ptr<std::vector<char>> Managed buffer that keeps data alive
+   * 
+   * This utility function provides consistent buffer allocation and initialization
+   * for both event trace and firmware log reports. Uses the standard debug_buffer_size
+   * constant. The returned unique_ptr ensures automatic memory management and should 
+   * be kept alive while using log_buffer.
```

Comment:
This is not really necessary.  The function returns a std::vector<>, as long as the std::vector is kept alive, the log buffer is alive.  Can you return simply std::vector